src/Reflector/ClassReflector.php

```diff
@@ -32,11 +36,15 @@ public function __construct(SourceLocator $sourceLocator)
      */
     public static function buildDefaultReflector() : self
     {
-        return new self(new AggregateSourceLocator([
-            new PhpInternalSourceLocator(),
-            new EvaledCodeSourceLocator(),
-            new AutoloadSourceLocator(),
-        ]));
+        $locator = new PhpParserLocator(
+            new MemoizingParser((new ParserFactory())->create(ParserFactory::PREFER_PHP7))
+        );
+
+        return new self(new MemoizingSourceLocator(new AggregateSourceLocator([
```

Comment:
Memoizing locator results removes the need for repeated I/O operations. A single memoizing locator around all the code should be sufficient